CharacterGenerator/CharacterGenerator/CreateSoldierTableViewController.m

```diff
@@ -0,0 +1,317 @@
+//
+//  CreateSoldierTableViewController.m
+//  CharacterGenerator
+//
+//  Created by Z on 8/2/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "CreateSoldierTableViewController.h"
+#import "COD4database.h"
+
+@interface CreateSoldierTableViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *nameTextField;
+
+@property (nonatomic) COD4database *model;
+
+@end
+
+@implementation CreateSoldierTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    //setup model
+    self.model = [[COD4database alloc] init];
+    [self.model setup];
+    
+    // Connect data
+    self.teamPicker.dataSource = self;
+    self.teamPicker.delegate = self;
+    
+    self.primaryWeaponTypePicker.dataSource = self;
+    self.primaryWeaponTypePicker.delegate = self;
+    
+    self.primaryWeaponPicker.dataSource = self;
+    self.primaryWeaponPicker.delegate = self;
+    
+    self.secondaryWeaponTypePicker.dataSource = self;
+    self.secondaryWeaponTypePicker.delegate = self;
+    
+    self.secondaryWeaponPicker.dataSource = self;
+    self.secondaryWeaponPicker.delegate = self;
+    
+    self.perk1Picker.dataSource = self;
+    self.perk1Picker.delegate = self;
+    
+    self.perk2Picker.dataSource = self;
+    self.perk2Picker.delegate = self;
+    
+    self.perk3Picker.dataSource = self;
+    self.perk3Picker.delegate = self;
+    
+}
+
+-(void)saveRecruit{
+    
+    Soldier *recruit = [[Soldier alloc] init];
+    recruit.name = self.nameTextField.text;
+    
+    NSInteger teamRow = [self.teamPicker selectedRowInComponent:0];
+    recruit.team = [[self.model.team allKeys] objectAtIndex:teamRow];
+    
+    NSInteger primaryWeaponTypeRow = [self.primaryWeaponTypePicker selectedRowInComponent:0];
+    recruit.primaryWeaponType = [[self.model.primaryWeapons allKeys] objectAtIndex:primaryWeaponTypeRow];
+    
+    NSInteger primaryWeaponRow = [self.primaryWeaponPicker selectedRowInComponent:0];
+    NSArray *primaryWeaponsOfType = [self.model.primaryWeapons objectForKey:recruit.primaryWeaponType];
+    recruit.primaryWeapon = [primaryWeaponsOfType objectAtIndex:primaryWeaponRow];
+    
+    NSInteger secondaryWeaponTypeRow = [self.secondaryWeaponTypePicker selectedRowInComponent:0];
+    recruit.secondaryWeaponType = [[self.model.secondaryWeapons allKeys] objectAtIndex:secondaryWeaponTypeRow];
+    
+    NSInteger secondaryWeaponRow = [self.secondaryWeaponPicker selectedRowInComponent:0];
+    NSArray *secondaryWeaponsOfType = [self.model.secondaryWeapons objectForKey:recruit.secondaryWeaponType];
+    recruit.secondaryWeapon = [secondaryWeaponsOfType objectAtIndex:secondaryWeaponRow];
+    
+    recruit.specialGrenade = [self.model.specialGrenades objectAtIndex:self.specialGrenadeSegmentedControl.selectedSegmentIndex];
```

Comment:
I've added a default grenade to our project as you suggested. It won't crash now. We noticed that when we were doing our project, we just tried to make sure we added grenade when we were demoing. It's great that this fixes the problem now. Thanks a lot, Eric.